vg/draw/canvas.go

```diff
@@ -9,14 +9,33 @@ import (
 	"image/color"
 	"math"
 	"strings"
+	"sync"
 
 	"gonum.org/v1/plot/vg"
-	"gonum.org/v1/plot/vg/vgeps"
-	"gonum.org/v1/plot/vg/vgimg"
-	"gonum.org/v1/plot/vg/vgpdf"
-	"gonum.org/v1/plot/vg/vgsvg"
 )
 
+// formats holds the registered canvas image formats
+var formats struct {
+	sync.Mutex
+	m map[string]func(w, h vg.Length) vg.CanvasWriterTo
+}
+
+// RegisterFormat registers an image format for use by NewformattedCanvas.
+// Name is the name of the format, like "jpeg" or "png".
+// New is the construction function to call for the format.
+func RegisterFormat(name string, fn func(w, h vg.Length) vg.CanvasWriterTo) {
+	formats.Lock()
+	defer formats.Unlock()
+	if formats.m == nil {
+		formats.m = make(map[string]func(w, h vg.Length) vg.CanvasWriterTo)
```

Comment:
couldn't we deal with the initialization of the map when creating the `formats` value?

Reply:
Done.